### ISSUE TITLE
Cluster-manager installation fails via baremetal provisioning route

### ISSUE THREAD

**Issue by rkharya** [contiv/cluster#80]
Following baremetal provisoinig readme - https://github.com/contiv/cluster/blob/master/management/baremetal.md

Took care of all required pre-requisites, clonned ansible from git repo to local host and created /tmp/host file as below -

[cluster-admin@contiv-aci-scale-1 ansible]$ cat /tmp/hosts
[cluster-control]
contiv-aci-scale-1 ansible_host=127.0.0.1

Running ansible play-book fails with error as below -

[cluster-admin@contiv-aci-scale-1 ansible]$ ansible-playbook --key-file=~/.ssh/id_rsa -i /tmp/hosts ./site.yml
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and make sure become_method is 'sudo' (default). This feature will be removed in a future release. Deprecation
 warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.

PLAY ***************************************************************************
skipping: no hosts matched

PLAY ***************************************************************************
skipping: no hosts matched

PLAY ***************************************************************************
skipping: no hosts matched

PLAY ***************************************************************************

TASK [setup] *******************************************************************
Wednesday 06 April 2016  03:23:05 -0700 (0:00:00.304)       0:00:00.304 *******
fatal: [contiv-aci-scale-1]: FAILED! => {"failed": true, "msg": "'env' is undefined"}

NO MORE HOSTS LEFT *************************************************************
	to retry, use: --limit @./site.retry

PLAY RECAP *********************************************************************
contiv-aci-scale-1         : ok=0    changed=0    unreachable=0    failed=1

Wednesday 06 April 2016  03:23:05 -0700 (0:00:00.146)       0:00:00.450
===============================================================================
TASK: setup ------------------------------------------------------------- 0.15s

Looks like its expecting 'env' variables to be defined. Let me know what are those and pls. have them updated in this doc, if any.

**Comment by vvb** [comment]
@rkharya you need to pass `env` in the extra-vars

**Comment by rkharya** [comment]
Do you mean we should be using like this - '{"env":{"http_proxy":"http://proxy.esl.cisco.com:8080","HTTP_PROXY":"http://proxy.esl.cisco.com:8080","https_proxy":"http://proxy.esl.cisco.com:8080","HTTPS_PROXY":"http://proxy.esl.cisco.com:8080", "no_proxy": "", "NO_PROXY": ""},"control_interface": "enp2s0f0","validate_certs": "false","docker_version": "1.10.3"}'

And can i specify Docker CS engine version here?

**Comment by rkharya** [comment]
So looks like "docker_version":"1.10.3.cs2" worked. We can close this issue once we update md file with following -
- environment variable setting required before running ansible-playbook
- cluster-admin user should be able to do passwordless ssh to the control node itself

**Comment by mapuri** [comment]
@rkharya thanks for verifying docker-cs version part!! I didn't know it was as simple :) 

> We can close this issue once we update md file with following
> environment variable setting required before running ansible-playbook

I think this is already documented here: https://github.com/contiv/cluster/blob/master/management/ansible_vars.md#mandatory-variables

> cluster-admin user should be able to do passwordless ssh to the control node itself

I think this is documented as well here (i.e. the passwordless sudo and adding key to authorized hosts): https://github.com/contiv/cluster/blob/master/management/baremetal.md#0-ensure-following-pre-requisites-are-met-on-the-control-host

Please let me know if you like these to reworded in some way. Feel free to submit a PR or give mthe change and I can do it for you.

**Comment by rkharya** [comment]
I think this is fine. We just need to specifically call-out that we need to define environmental variables based on ansible_vars.md. Since actual command was already provided, one can easily miss variable settings pre-req.

**Comment by mapuri** [comment]
i see, got it. So I need to fix the ansible-playbook [command here](https://github.com/contiv/cluster/blob/master/management/baremetal.md#1-download-and-install-cluster-manager-on-the-control-host) to pass `env`. 

I will take care of this and we can close this PR once that is done. Thanks again @rkharya for trying this out.

**Comment by rkharya** [comment]
*** we might need below changes in repos to install docker_version=1.10.3.cs2 -

As part of regular commissioning we get docker-main.repo with base URL as -
baseurl=https://yum.dockerproject.org/repo/main/centos/7

we should modify it as -
baseurl=https://packages.docker.com/1.10/yum/repo/main/centos/7

**Comment by mapuri** [comment]
@rkharya 

> we might need below changes in repos to install docker_version=1.10.3.cs2

hmm, so docker cs version installation needs this additional change? Or is it optional? If not, can you please track this in `contiv/ansible` repo? I had closed an open issue https://github.com/contiv/ansible/issues/103 based on previous findings above. Feel free to reopen that issue or file a new one with these details as it will help us make ansible side changes accordingly.

**Comment by rkharya** [comment]
It doesn't seems to be optional. Since my host was already having that repo configured, simply pointing global var 'docker_version' to 1.10.3.cs2 worked on that host. Its failing on other nodes where repo is not configured. I have validated this by running failed task manually -

Apr 06 21:16:41 contiv-aci-scale-1.cisco.com clusterm[30861]: level=info msg="TASK [docker : install docker (redhat)] ****************************************"
Apr 06 21:16:46 contiv-aci-scale-1.cisco.com clusterm[30861]: level=info msg="fatal: [contiv-aci-scale-2-FCH1924V2EK]: FAILED! => {\"changed\": true, \"cmd\": \"curl https://get.docker.com | sed 's/docker-engine/docker-engine-1.10.3.cs2/' | bash\", \"delta\": \"0:00:03.985444\", \"end\": \"2016-04-06 21:16:46.152347\", \"failed\": true, \"rc\": 1, \"start\": \"2016-04-06 21:16:42.166903\", \"stderr\": \"  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\\n                                 Dload  Upload   Total   Spent    Left  Speed\\n\\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\\r100 14028  100 14028    0     0  76725      0 --:--:-- --:--:-- --:--:-- 77076\\n+ sh -c 'sleep 3; yum -y -q install docker-engine-1.10.3.cs2'\\nError: Nothing to do\", \"stdout\": \"\", \"stdout_lines\": [], \"warnings\": [\"Consider using get_url module rather than running curl\"]}"

Ansible installed docker-main.repo doesn't understand docker-cs version -

[cluster-admin@contiv-aci-scale-1 ~]$ cat /etc/yum.repos.d/docker-main.repo
[docker-main-repo]
name=Docker main Repository
baseurl=https://yum.dockerproject.org/repo/main/centos/7
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg

Filing a new issue for this under contiv/ansible.
